### PR TITLE
Show a warning and block access to the plugin if disk space is low

### PR DESCRIPTION
**Pull request by willmot**
We now detect if disk space is low, if it is we block access to the plugin and show a warning, which looks like this:

screen shot 2016-03-12 at 17 46 27

This PR fixes: https://github.com/humanmade/backupwordpress/issues/615

### CODE REVIEW

**Comment by pdewouters** [comment]
I thought that function wasn't reliable? https://github.com/humanmade/backupwordpress/issues/615

**Comment by willmot** [comment]
`disk_free_space` is safe, it's `disk_total_space` which isn't

**Comment by willmot** [comment]
Added error checking when setting the number of backups to keep to ensure users don't set the plugin to store more backups than they have disk space available.

screen shot 2016-03-13 at 12 35 46

See #907

File: admin/actions.php [hunk 1]
```diff
@@ -351,6 +352,10 @@ function edit_schedule_submit() {
 			$errors['hmbkp_schedule_max_backups'] = __( 'Max backups must be greater than 0', 'backupwordpress' );
 		}
 
+		elseif ( $site_size->is_site_size_cached() && disk_space_low( $site_size->get_site_size() * $max_backups ) ) {
+			$errors['hmbkp_schedule_max_backups'] = sprintf( __( 'Storing %s backups would use %s of disk space but your server only has %s free.', 'backupwordpress' ), '<code>' . number_format_i18n( $max_backups ) . '</code>', '<code>' . size_format( $max_backups * $site_size->get_site_size() ) . '</code>', '<code>' . size_format( disk_free_space( Path::get_path() ) ) . '</code>' );
```
Comment (pdewouters) [inline admin/actions.php]: A comment might be useful for translators, e.g. `/* translators: first placeholder is number of backups, second is free space available */`

File: functions/interface.php [hunk 2]
```diff
@@ -191,6 +191,10 @@ function set_server_config_notices() {
 		$messages[] = sprintf( __( 'Your database cannot be backed up because your server doesn\'t support %1$s or %2$s. Please contact your host and ask them to enable them.', 'backupwordpress' ), '<code>mysqldump</code>', '<code>PDO</code>' );
 	}
 
+	if ( disk_space_low() ) {
+		$messages[] = sprintf( __( 'You\'re server only has %s of disk space left which probably isn\'t enough to complete a backup. Try deleting some existing backups or other files to free up space.', 'backupwordpress' ), '<code>' . size_format( disk_free_space( Path::get_path() ) ) . '</code>' );
```
Comment (pdewouters) [inline functions/interface.php]: should be "your" not "you're"

Comment (willmot) [inline functions/interface.php]: your right :-)

**Comment by pdewouters** [comment]
needs master merging

**Comment by willmot** [comment]
@pdewouters merged master and fixed the typo and translator comment 👍

**Pull request merged**